sdk/python/lib/pulumi/runtime/rpc.py

```diff
@@ -598,7 +600,8 @@ def deserialize_resource(
         ref_id = ref_struct["id"]
         return deserialize_property(UNKNOWN if ref_id == "" else ref_id, keep_unknowns)
 
-    return urn
+    res = Resource(urn, "urn", False)
```

Comment:
This is actually changing behavior. We may want to revisit the behavior, but the current intention is to fallback to returning the `urn`. Can we add a lint ignore instead? Or change the type signature of `deserialize_resource` to `Union["Resource", str]` (`Union` will have to be imported from `typing`).

Reply:
I tried simply using `cast(“Resource, urn)` but the type checker still complained. I’ll try the latter first